.github/workflows/contrib-tests.yml

```diff
@@ -22,7 +22,17 @@ permissions:
   # contents: read
   # deployments: read
 jobs:
+  DeployPostgres:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install PostgreSQL
+        run: |
+          sudo apt install postgresql -y
+      - name: Start PostgreSQL service
+        run: sudo service postgresql start
+
   RetrieveChatTest:
+    needs: DeployPostgres
```

Comment:
I'm not sure about this approach. How about making "DeployPostgres" one step inside the RetrieveChatTest job and keep the "runs-on" condition?

Reply:
I have done as you recommended.